airbyte-ci/connectors/metadata_service/lib/metadata_service/gcs_upload.py

```diff
@@ -237,6 +302,31 @@ def _apply_author_info_to_metadata_file(metadata_dict: dict, original_metadata_f
     return metadata_dict
 
 
+def _apply_python_components_sha_to_metadata_file(
+    metadata_dict: dict,
+    python_components_sha256: Optional[Path] | None = None,
+) -> dict:
+    """If a `components.py` file is required, store the necessary information in the metadata.
+
+    This adds a `required=True` flag and the sha256 hash of the `python_components.zip` file.
+
+    This is a no-op if `python_components_sha256` is not provided.
+    """
+    if python_components_sha256:
+        metadata_dict = set_(
```

Comment:
To have that sha in metadata, you'd need to also change the metadata schema. @alafanechere pointed me at where it is, but of course I already forgot.

Reply:
Found it: https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/metadata_service/lib/metadata_service/models/src/ConnectorTestSuiteOptions.yaml#L10